source/io.h

```diff
@@ -882,11 +882,13 @@ class source
     )
         -> bool
     {
-        std::ifstream in{ filename };
-        if (!in.is_open()) {
+        //  If filename is stdin, we read from stdin, otherwise we try to read the file
+        //
+        std::ifstream fss{ filename };
```

Comment:
Maybe a little more explicit, only compare once, and don't try to open `stdin`.

```
auto is_stdin = filename == "stdin";
std::ifstream fss;
if (!is_stdin) {
    fss.open(filename);

    if (!fss.is_open()) {
        return false;
    }
}
std::istream& in = is_stdin ? std::cin : fss;

auto in_comment            = false;
```

Reply:
Note: other than the "don't try to open stdin" the rest is personal preference, so just suggestions, feel free to do it differently.

Reply:
Ah gotcha, I missed that. Just pushed a fix.

> Then again, maybe it would be expected to read that file, and stdin should be an explicit option rather than using the string `stdin` in the filename place.

Yea, I was thinking about this too. When I originally attempted this, I created a cli flag which would turn on `stdin` mode. But I realized cppfront is using this similar logic with the `-o` flag; if you specify `-o stdout`, it will output to stdout instead of file, so I figured it was following convention to use the same logic for file reading

Reply:
Makes sense to be consistent for now.